acc_acc_modules/1_annotate_acc_modules.py

```diff
@@ -0,0 +1,373 @@
+pa14_gene_annot = pd.read_csv(pa14_gene_annot_filename, index_col=0, header=0)
+# -
+
+# Import metadata of samples
+metadata_filename = paths.SAMPLE_METADATA
+
+pao1_gene_annot = pao1_gene_annot["Name"].to_frame("gene name")
+pa14_gene_annot = pa14_gene_annot["Name"].to_frame("gene name")
+
+# ## Add gene names
+
+# Add gene names
+pao1_gene_module_labels = pao1_membership.merge(
+    pao1_gene_annot, left_index=True, right_index=True
+)
+pa14_gene_module_labels = pa14_membership.merge(
+    pa14_gene_annot, left_index=True, right_index=True
+)
+
+print(pao1_gene_module_labels.shape)
+pao1_gene_module_labels.head()
+
+print(pa14_gene_module_labels.shape)
+pa14_gene_module_labels.head()
+
+# ## Add expression information
+#
+# 1. What is the baseline level of expression for each gene in the module?
+# 2. What is the expression level of genes in a clinical context (i.e. clinical samples)?
+
+# Read in expression data
+pao1_compendium = pd.read_csv(paths.PAO1_COMPENDIUM, sep="\t", index_col=0)
+pa14_compendium = pd.read_csv(paths.PA14_COMPENDIUM, sep="\t", index_col=0)
+
+pao1_compendium.head()
+
+# Calculate median expression across all samples
+pao1_median_all = pao1_compendium.median().to_frame("median expression")
+pa14_median_all = pa14_compendium.median().to_frame("median expression")
+
+pao1_median_all.head()
+
+
+# Select subset of samples and calculate the median expression across that subset of samples
+# TO DO: Move this into utils
+def get_sample_ids(
+    metadata_filename, experiment_colname, sample_colname, experiment_id
+):
+    """
+    Returns sample ids (found in gene expression df) associated with
+    a given list of experiment ids (found in the metadata)
+
+    Arguments
+    ----------
+    metadata_filename: str
+        File containing metadata
+    experiment_colname: str
+        Column header that contains experiment id that maps expression data
+        and metadata
+    sample_colname: str
+        Column header that contains sample id that maps expression data
+        and metadata
+    experiment_id: str
+        Selected experiment id to grab samples from
+
+    """
+    # Read in metadata
+    metadata = pd.read_csv(metadata_filename, header=0)
+    metadata.set_index(experiment_colname, inplace=True)
+
+    selected_metadata = metadata.loc[experiment_id]
+    sample_ids = list(selected_metadata[sample_colname])
+
+    return sample_ids
+
+
+# +
+# TO DO: Have Deb or Georgia select a study
+# Looks like we removed many of the clinical isolates from this compendium with our strain binning
+# selected_sample_ids = get_sample_ids(
+#   metadata_filename, experiment_colname="SRA_study", sample_colname="Experiment", experiment_id="SRP063289")
+
+# +
+# Subset compendium
+# subset_pao1_compendium = pao1_compendium.loc[selected_sample_ids]
+# subset_pa14_compendium = pa14_compendium.loc[selected_sample_ids]
+
+# +
+# print(subset_pao1_compendium.shape)
+# print(subset_pa14_compendium.shape)
+
+# +
```

Comment:
Just want to confirm that the notebook will run without errors with these sections commented out. I'm sure it's fine, I've just run into problems with that before.

Reply:
Yes this code allows me to select a subset of samples of interest to get expression from, but I couldn't find a subset that I had expression data for yet, so they are commented out for now in case we are able to find a subset after talking with my collaborators. I will make a note of this